src/Extensions/Hosting.Services/ServiceFabricHostBuilder.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Fabric;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services
+{
+	/// <summary>
+	/// Wrapper on top of IHostBuilder to propagete proper context type and avoid type registration mistakes
+	/// </summary>
+	public sealed class ServiceFabricHostBuilder<TContext> where TContext : ServiceContext
+	{
+		/// <summary>
+		/// Method should be called only by extensions of this class
+		/// If you are creating service please register dependencies before colling BuildService
```

Comment:
Not that I can think about, at least without making code uglier.
I'll think more about it, but I don't think that it's critical since it won't influence code execution.

Reply:
I've looked more into it and it looks like this method could be done internal, so it should ensure proper usage